internal/mode/static/telemetry/collector.go

```diff
@@ -0,0 +1,99 @@
+package telemetry
+
+import (
+	"context"
+
+	v1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/nginxinc/nginx-gateway-fabric/internal/mode/static/state/graph"
+)
+
+type DataCollector interface {
+	Collect(ctx context.Context) Data
+}
+
+type GraphGetter interface {
+	GetLatestGraph() *graph.Graph
+}
+
+type GraphResourceCount struct {
+	Gateways       int
+	GatewayClasses int
+	HTTPRoutes     int
+	Secrets        int
+	Services       int
+}
+
+type ProjectNameAndVersion struct {
+	Name    string
+	Version string
+}
+
+// Data is telemetry data.
+// Note: this type might change once https://github.com/nginxinc/nginx-gateway-fabric/issues/1318 is implemented.
+type Data struct {
+	ProjectNameAndVersion ProjectNameAndVersion
+	NodeCount             int
+	GraphResourceCount    GraphResourceCount
+}
+
+type DataCollectorImpl struct {
+	k8sClientReader client.Reader
+	graphGetter     GraphGetter
+	version         string
+}
+
+func NewDataCollector(
+	k8sClientReader client.Reader,
+	graphGetter GraphGetter,
+	version string,
+) *DataCollectorImpl {
+	return &DataCollectorImpl{
+		k8sClientReader: k8sClientReader,
+		graphGetter:     graphGetter,
+		version:         version,
+	}
+}
+
+func (c DataCollectorImpl) Collect(ctx context.Context) Data {
+	nodeCount := collectNodeCount(ctx, c.k8sClientReader)
+	graphResourceCount := collectGraphResourceCount(c.graphGetter)
+
+	data := Data{
+		NodeCount:          nodeCount,
+		GraphResourceCount: graphResourceCount,
+		ProjectNameAndVersion: ProjectNameAndVersion{
+			Name:    "NGF",
+			Version: c.version,
+		},
+	}
+
+	return data
+}
+
+func collectNodeCount(ctx context.Context, k8sClient client.Reader) int {
+	nodes := v1.NodeList{}
+	_ = k8sClient.List(ctx, &nodes)
```

Comment:
Are we intentionally ignoring errors @pleshakov?

Reply:
Ah I forgot, I was going to ask about errors, wasn't sure the best way to handle them outside of printing an error through a Logger. Do we still want to return Data if some of the fields ran into errors? Should we retry up to a certain attempts?

Reply:
> Are we intentionally ignoring errors @pleshakov?

nope. but they are not that important for the draft 

> Ah I forgot, I was going to ask about errors, wasn't sure the best way to handle them outside of printing an error through a Logger. Do we still want to return Data if some of the fields ran into errors? Should we retry up to a certain attempts?

Collect() should return an error back to the caller. Retries can be configured in the telemetry job if necessary.